include/rsl/monad.hpp

```diff
@@ -36,7 +39,8 @@ template <typename T, typename Fn>
  * @return     The return type of the function
  */
 template <typename T, typename E, typename Fn>
-[[nodiscard]] constexpr auto mbind(tl::expected<T, E> const& exp, Fn fn) {
+[[nodiscard]] constexpr auto mbind(tl::expected<T, E> const& exp, Fn fn)
+    -> std::invoke_result_t<Fn, T> {
```

Comment:
Why add this explicit return type back?

Reply:
It fails to build if you remove this. We weren't testing it before.